lib/controllers/widget/custom_fields_card.dart

```diff
@@ -0,0 +1,320 @@
+}
+
+class _CustomFieldsCardState extends State<CustomFieldsCard>
+    with TickerProviderStateMixin<CustomFieldsCard> {
+  bool _isExpanded = false;
+  @override
+  Widget build(BuildContext context) {
+    return new Column(children: <Widget>[
+      new AnimatedSize(
+          vsync: this,
+          duration: const Duration(milliseconds: 500),
+          child: new ConstrainedBox(
+              constraints: new BoxConstraints(),
+              child:
+                  _isExpanded ? customFull(context) : customSummary(context)))
+    ]);
+  }
+
+  /// Collapsed card widget to show a breif summary of the custom fields of a
+  /// device.
+  Widget customSummary(BuildContext context) => Card(
+        child: Column(
+          children: <Widget>[
+            Row(
+              mainAxisAlignment: MainAxisAlignment.start,
+              mainAxisSize: MainAxisSize.max,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: [
+                Container(
+                  margin: EdgeInsets.only(left: 10),
+                  width: MediaQuery.of(context).size.width * 0.4,
+                  child: Text(
+                    'Custom fields',
+                    style: TextStyle(
+                        color: Colors.black,
+                        fontSize: 28,
+                        fontWeight: FontWeight.w500),
+                    overflow: TextOverflow.clip,
+                  ),
+                ),
+                Container(
+                    height: MediaQuery.of(context).size.height * 0.3,
+                    width: 0,
+                    margin: EdgeInsets.all(0),
+                    child: VerticalDivider(
+                      thickness: 1,
+                    )),
+                Container(
+                    width: MediaQuery.of(context).size.width * 0.4,
+                    margin: EdgeInsets.only(left: 25.0),
+                    child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: <Widget>[
+                        Padding(
+                            padding: EdgeInsets.only(top: 10, bottom: 10),
+                            child: Column(
+                              crossAxisAlignment: CrossAxisAlignment.start,
+                              children: <Widget>[
+                                Text(
+                                  'Asset ID',
+                                  style: TextStyle(
+                                      fontSize: 18,
+                                      color: Colors.black87,
+                                      fontWeight: FontWeight.w500),
+                                  overflow: TextOverflow.clip,
+                                ),
+                                Text(
+                                  widget._device.annotatedAssetId == null
+                                      ? '-'
+                                      : widget._device.annotatedAssetId,
```

Comment:
For all of these instead of using the same logic use a function getValueOrDefault(widget._device.property)

Reply:
Done.

Reply:
Actually, should do this using the ?? operator.  ie: widget._device.annotatedAssertId ?? '-'  don't use a function ;)

Reply:
Got it :)